source/testing/testing-routes.md

```diff
@@ -35,54 +35,57 @@ export default Route.extend({
-code into smaller chunks (or "concerns"), allows it to be more readily isolated 
+The action `displayAlert` contains the code that is called when the action is
+received, and the private function `_displayAlert` performs the work. While not
+necessarily obvious here because of the small size of the functions, separating
+code into smaller chunks (or "concerns"), allows it to be more readily isolated
 for testing, which in turn allows you to catch bugs more easily.
 
-Here is an example of how to unit test this route:
+Here is an example of test this route in an isolated test case:
 
 ```tests/unit/routes/application-test.js
-import { moduleFor, test } from 'ember-qunit';
+import { module, test } from 'qunit';
+import { setupTest } from 'ember-qunit';
 
 let originalAlert;
 
-moduleFor('route:application', 'Unit | Route | application', {
-  beforeEach() {
+module('Unit | Route | application', function(hooks) {
+  setupTest(hooks);
+
+  hooks.beforeEach(function(assert) {
     originalAlert = window.alert; // store a reference to window.alert
-  },
+  });
 
-  afterEach() {
+  hooks.afterEach(function(assert) {
     window.alert = originalAlert; // restore window.alert
-  }
-});
+  });
+
+  test('should display an alert', function(assert) {
+    assert.expect(2);
+
+    // with moduleFor, the subject returns an instance of the route
+    let route = this.owner.lookup('route:application');
+
+    // stub window.alert to perform a qunit test
+    const expectedTextFoo = 'foo';
+    window.alert = (text) => {
+      assert.equal(text, expectedTextFoo, `expect alert to display ${expectedTextFoo}`);
+    };
+
+    // call the _displayAlert function which triggers the qunit test above
+    route._displayAlert(expectedTextFoo);
+
+    // set up a second stub to perform a test with the actual action
+    const expectedTextBar = 'bar';
+    window.alert = (text) => {
+      assert.equal(text, expectedTextBar, `expected alert to display ${expectedTextBar}`);
+    };
 
-test('should display an alert', function(assert) {
-  assert.expect(2);
-
-  // with moduleFor, the subject returns an instance of the route
-  let route = this.subject();
-
-  // stub window.alert to perform a qunit test
-  const expectedTextFoo = 'foo';
-  window.alert = (text) => {
-    assert.equal(text, expectedTextFoo, `expect alert to display ${expectedTextFoo}`);
-  };
-
-  // call the _displayAlert function which triggers the qunit test above
-  route._displayAlert(expectedTextFoo);
-
-  // set up a second stub to perform a test with the actual action
-  const expectedTextBar = 'bar';
-  window.alert = (text) => {
-    assert.equal(text, expectedTextBar, `expected alert to display ${expectedTextBar}`);
-  };
-    
-  // Now use the routes send method to test the actual action
-  route.send('displayAlert', expectedTextBar);
+    // Now use the routes send method to test the actual action
+    route.send('displayAlert', expectedTextBar);
+  });
```

Comment:
IMHO this should be two separate tests, though I don't really see the necessity here for having two different things anyway 🤔